Automattic/wp-calypso
# Gutenberg: Load the selected editor preference as soon as possible

#### Changes proposed in this Pull Request

* Create a new `QuerySiteSelectedEditor` query component that requests the selected editor preference for a given site. Also create the required data layer wirings.
* Add the component to `<Layout>`, in order to fetch the selected editor preference as soon as possible.

#### Testing instructions

* Open the Network tab in the browser's developer tools, and filter it for XHR requests with keyword "gutenberg".
* Load Calypso, and make sure a request to `/gutenberg` is fired.
* Input `state.selectedEditor[ state.ui.selectedSiteId ]` and make sure it has a defined value (`classic` or `gutenberg`).
* Now change site.
* Observe that a new `/gutenberg` request is fired, and `state.selectedEditor[ state.ui.selectedSiteId ]` for the new site has a defined value too.
